### ISSUE TITLE
conda install: package not found - using miniforge distribution on Apple ARM

### ISSUE THREAD

**Issue by clstaudt** [networkit/networkit#771]
[Miniforge](https://github.com/conda-forge/miniforge) is my preferred Anaconda-like distribution for  getting the Python data science stack to work on Apple ARM. 

Trying to install networkit:

```
╰─ conda install -c conda-forge networkit                                                                                                              (base) 
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - networkit

Current channels:

  - https://conda.anaconda.org/conda-forge/osx-arm64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://repo.anaconda.com/pkgs/main/osx-arm64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/osx-arm64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.

```

Is this due to missing support for ARM?

**Comment by avdgrinten** [comment]
Probably. This should be fixed.

Does NetworKit compile from source on Apple M1?

Not sure how much work it is to iron out ARM issues (and to add a Linux/Apple ARM builds to the CI) but it's probably not too bad. This can be done after the next release is out.

**Comment by fabratu** [comment]
I think the feedback is expected, since arm64 support for packages has to be enabled [explicitely](https://conda-forge.org/blog/posts/2020-10-29-macos-arm64/). Currently conda packages are only build for [x86_64](https://anaconda.org/conda-forge/networkit/files).

Unfortunately we currently have no M1-device available. Also Github Actions currently does not support arm64-runners at all (due to the underlying Azure instances). However, after the next release we should try to use the conda-infrastructure to see whether the packages can be build successfully.

**Comment by avdgrinten** [comment]
In principle, we can either use a different CI provider for aarch64 (in addition to GitHub actions), or we can cross-compile NetworKit for aarch64 (that should be relatively easy) and test it using a Linux VM such as user-mode qemu (I don't know how easy that is).

This does not really work for Apple M1 due to Apple's licensing politics though. But getting the aarch64 build to work on Linux would be a good first step towards M1 support.

**Comment by avdgrinten** [comment]
Apparently Big Sur can cross-compile to aarch64, but then we still have no way to test the resulting binaries on M1 (GitHub actions lacks the runners and so do other free CI providers).

**Comment by fabratu** [comment]
One possibility for aarch64 (Linux-variant) would be using aarch-manylinux Docker-environments. For wheel-builders it seems quite common to include such setups, like in [cibuildwheel](https://github.com/pypa/cibuildwheel/blob/main/cibuildwheel/linux.py#L72). Many common python-packages therefore also provide aarch64 packages (like [matplotlib](https://github.com/matplotlib/matplotlib/blob/master/.github/workflows/cibuildwheel.yml#L28)). It should also be possible to set something up manually with reasonable effort. 

One workaround for now would be to use the brew-bottle. They automatically create bottles for arm64 (including NetworKit) - likely based on cross-compiling on their macOS runners. In general this also implies, that the binary can be build and has at least support for creating graphs and adding edges. At least that is normally tested for brew-packages.

**Comment by fabratu** [comment]
I added arm64-darwin builds for our packages. While the dependent packages (libtlx, libnetworkit) are already merged, networkit update is still a PR since it needs testing on both arm64-darwin and aarch64. The package is available as [artifact](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=365200&view=artifacts&pathAsName=false&type=publishedArtifacts) and I tested it on a M1 Macbook Pro.  

@clstaudt Is it possible for you to download and install the osx-arm64 package, so that we have a second test device? 

The package-artifact itself is in subfolder `osx-arm64` of the archive (named: `networkit-9.0-py39h7f752ed_1.tar.bz2`) Since this doesn't install all dependencies, the following commands should lead to a successful installation:

```
conda install <PATH>/networkit-9.0-py39h7f752ed_1.tar.bz2
conda install libnetworkit
(maybe) conda install numpy
```
